### ISSUE TITLE
List not handled gracefully

### ISSUE THREAD

**Issue by andysturrock** [drewnoakes/dasher#11]
The following code throws a `System.Security.VerificationException : Operation could destabilize the runtime.`:

```c#
public class WithList
{
    public WithList(List<int> list)
    {
        ListOfInt = list;
    }
    public List<int> ListOfInt { get; }
}

void BlowUp()
{
    List<int> list = new List<int>();
    var msg = new WithList(list);
    new Serialiser<WithList>().Serialise(msg);
}

```

I'll take a look but you might get there quicker :-)

**Comment by drewnoakes** [comment]
List should not be supported as it's mutable. Use IReadOnlyList instead.
However there is a bug here: it should throw on creating the serialiser.

**Comment by andysturrock** [comment]
I just forked the repo to look at this.  It is now more graceful as it throws `System.Exception : Unable to serialise type System.Collections.Generic.List`1[System.Int32]`.  Your NuGet release is behind the latest code in GibHub I guess.

**Comment by drewnoakes** [comment]
Yes that is inconsistent. Will have a think. I like immutability and it's important to make it possible, but maybe Dasher shouldn't force it on users. Will have a think.

Just pushed 0.3.0 to NuGet.
